dcos/dcos-ui
# DCOS-48338: add placement form section

Add placement form section to jobs form.

## Testing

Using an OSS cluster:
- Go to jobs form > Placement tab
- Try adding a placement constraint to a job (it should work)
- Try submitting with an incomplete constraint/some fields left unfilled (you should see validation messages)
- Try adding a constraint in the form and then changing it in the JSON editor so that `constraints` is no longer an array (you should see validation message)

## Trade-offs



## Dependencies

This is the OSS implementation of the placement section - [DCOS-48338](https://jira.mesosphere.com/browse/DCOS-48338) is not complete without the EE implementation as well.

## Screenshots

<img width="988" alt="Screen Shot 2019-04-17 at 11 45 27 AM" src="https://user-images.githubusercontent.com/19582796/56313031-6d1fa800-6106-11e9-9dfb-c81f306fe3b9.png">
<img width="1135" alt="Screen Shot 2019-04-17 at 11 45 52 AM" src="https://user-images.githubusercontent.com/19582796/56313037-727cf280-6106-11e9-94ef-1b616bbf2862.png">
